src/services/dependencies.service.js

```diff
@@ -1,19 +1,48 @@
 // @flow
+import { PACKAGE_MANAGER_CMD } from './platform.services';
 const childProcess = window.require('child_process');
 
+/* TODO: Improve flow type for executor - no type annotation found
+         from facebook flow source https://github.com/facebook/flow/blob/v0.76.0/lib/core.js#L584:
+          constructor(callback: (
+                resolve: (result: Promise<R> | R) => void,
+                reject: (error: any) => void
+              ) => mixed): void;
+*/
+
+const spawnProcess = (
+  cmd: string,
+  cmdArgs: Array<string>,
+  projectPath: string,
+  [resolve, reject]: Array<any>
+) => {
+  const child = childProcess.spawn(cmd, cmdArgs, {
+    cwd: projectPath,
+  });
+
+  child.on('exit', code => {
+    if (code) {
+      reject(child.stderr);
+    } else {
+      resolve(child.stdout);
+    }
+  });
+
+  return child;
+};
+
 export const installDependency = (
   projectPath: string,
   dependencyName: string,
   version: string
 ) => {
-  return new Promise((resolve, reject) => {
-    // TODO: yarn?
-    childProcess.exec(
-      `npm install ${dependencyName}@${version} -SE`,
-      { cwd: projectPath },
-      (err, res) => {
-        err ? reject(err) : resolve(res);
-      }
+  return new Promise(function(resolve, reject) {
+    // no arrow function so we can use function arguments to pass
+    const child = spawnProcess(
+      PACKAGE_MANAGER_CMD,
+      ['add', `${dependencyName}@${version}`, '-SE'],
+      projectPath,
+      arguments
```

Comment:
So I've been thinking about this pattern ever since you brought it up in Gitter - I think it unncessarily complicates the code and would be a headache down the road if we ever need to add addtional arguments to `childProcess.spawn`. What about this instead?

```js
return new Promise((resolve, reject) => {
  const child = childProcess.spawn(
    PACKAGE_MANAGER_CMD,
    ['add', `${dependencyName}@${version}`, '-SE'],
    { cwd: projectPath }
  )
  child.on('exit', code => code ? reject(child.stderr) : resolve(child.stdout))
})
```

I think this is more clear as to its purpose and thus more maintainable long-term.

Reply:
I haven't tested it on my machine, but I'm not sure passing a completed stream will be very useful (`childProcess.spawn` returns streams, not buffers).

Reply:
@superhawk610 yes, you're right. I also don't like to pass the `resolve, reject` down to `spawn process`. Your code is working and I first thought I need to have access to `child` outside of promise but I can also add the logger after `child.on('exit')` handler.

I'll change this.

@bennygenel no problem. If you like I could create a new `window-support` branch in guppy repository that is based on master and merge your branch & `npm-to-yarn` branch into this new branch. Just not sure how we could handle this with your open PR. But we could probably create a new PR & close the previous.
Just if you don't have time to rebase your branch.